db/views.c

```diff
@@ -2519,13 +2519,15 @@ void views_unlock(void)
  * Get the name of the newest shard
  *
  */
-char *timepart_newest_shard(const char *view_name, unsigned long long *version)
+char *timepart_newest_shard(const char *view_name, unsigned long long *version,
+                            int lock)
 {
     timepart_views_t *views = thedb->timepart_views;
     timepart_view_t *view;
     char *ret = NULL;
 
-    Pthread_rwlock_rdlock(&views_lk);
+    if (lock)
```

Comment:
Is there a problem with nested read-locks?

Reply:
No, but read somewhere that it isn't considered a good practice.

Reply:
I find the extra code no needed

Reply:
I think it's needed. If we allow timepart_newest_shard() unlock the rdlock, I'll have to acquire it again in retrieve_schema().